### PR TITLE
[SYCL] Add test for sycl::vec with std::byte

### PR DESCRIPTION
**Pull request by maximdimakov**
In https://github.com/intel/llvm/pull/4637 added support for std::byte in sycl::vec class. The test is aimed to check functionality of sycl::vec class with std::byte.
Signed-off-by: mdimakov maxim.dimakov@intel.com

### CODE REVIEW

**Review by vladimirlaz — DISMISSED** [review]

File: SYCL/Basic/vector_byte.cpp [hunk 1]
```diff
@@ -0,0 +1,299 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
```
Comment (s-kanaev) [inline SYCL/Basic/vector_byte.cpp]: Should C++17 be explicitly selected here?

**Review by s-kanaev — DISMISSED** [review]

File: SYCL/Basic/vector_byte.cpp (hunk 1 above)
Comment (maximdimakov) [reply SYCL/Basic/vector_byte.cpp]: I added option to this line

**Comment by maximdimakov** [comment]
precommit is failing due to problems that are not related to this patch. Fix in the https://github.com/intel/llvm-test-suite/pull/494

**Comment by vladimirlaz** [comment]
@maximdimakov please note that  SYCL :: Basic/vector_byte.cpp has failed on CUDA

**Comment by maximdimakov** [comment]
@vladimirlaz I fixed problem with cuda

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**